### PR TITLE
Make MultipleSectionsAssigner a function component

### PR DESCRIPTION
**Pull request by megcrenshaw**
Paired with Kaitie to make the MultipleSectionsAssignment a function component. I did no refactors, except for removing two unused props: `forceReload` and `selectedSectionId`. I don't see props spreading to other components, so this shouldn't change current behavior.

The MultipleSectionsAssigner will be used on the Curriculum Catalog page. The dialog itself needs a refactor so that it's keyboard navigable, which is the next task. I plan on using a similar pattern that Jim used https://github.com/code-dot-org/code-dot-org/pull/51326/files, using the `FocusTrap` library.

## Testing story

Kaitie already added a bunch of unit tests for this component, and those tests caught a bug with the original refactor of the `selectAllHandler`! At this point, the fact those are passing is a good indicator that behavior didn't change. I did add a test for the teacher-facing courses and sections.

I also tested manually, adding multiple sections and using the "Select All" option and clicking the boxes manually.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/test/unit/templates/MultipleSectionAssignerTest.js [hunk 1]
```diff
@@ -151,29 +150,61 @@ describe('MultipleSectionsAssigner', () => {
       courseVersionId: assignedCourseANDUnitSection.courseVersionId,
     });
 
-    // Check that courses 1-6 have TeacherOptions and course 7 does not.
-    for (let i = 0; i < wrapper.instance().props.sections.length; i++) {
-      wrapper.instance().props.sections[i].participantType ===
```
Comment (megcrenshaw) [inline apps/test/unit/templates/MultipleSectionAssignerTest.js:156]: Unfortunately, the `instance()` method returns `null` for function components. https://enzymejs.github.io/enzyme/docs/api/ShallowWrapper/instance.html

File: apps/test/unit/templates/MultipleSectionAssignerTest.js [hunk 2]
```diff
@@ -151,29 +150,61 @@ describe('MultipleSectionsAssigner', () => {
...
-            wrapper
-              .find('TeacherSectionOption')
-              .filterWhere(
-                n =>
-                  n.props().section.id ===
-                  wrapper.instance().props.sections[i].id
-              )
-          ).to.have.lengthOf(0);
-    }
+    const assignableSections = fakeTeacherSectionsForDropdown.filter(
```
Comment (megcrenshaw) [inline apps/test/unit/templates/MultipleSectionAssignerTest.js:153]: We can't access the props of the component like we used to; we only have access to the props placed on the root node. See https://enzymejs.github.io/enzyme/docs/api/ShallowWrapper/props.html for more about this. As such, I couldn't access all the sections via the `sections` prop.

File: apps/test/unit/templates/MultipleSectionAssignerTest.js [hunk 3]
```diff
@@ -140,8 +140,7 @@ describe('MultipleSectionsAssigner', () => {
     ).to.be.false;
   });
 
-  // this feels like a spot check - should we check another version where the participant audience is 'teacher'?
-  it('renders all assignable sections for the course', () => {
+  it('renders all student sections for a student course', () => {
```
Comment (kobryan0619) [inline apps/test/unit/templates/MultipleSectionAssignerTest.js:143]: I appreciate this re-name.

File: apps/test/unit/templates/MultipleSectionAssignerTest.js [hunk 4]
```diff
@@ -151,29 +150,61 @@ describe('MultipleSectionsAssigner', () => {
...
+    notAssignableSections.forEach(section => {
+      expect(
+        wrapper
+          .find('TeacherSectionOption')
+          .filterWhere(option => option.props().section.id === section.id)
+      ).to.have.lengthOf(0);
+    });
+  });
+
+  it('renders all teacher sections for a teacher course', () => {
```
Comment (kobryan0619) [inline apps/test/unit/templates/MultipleSectionAssignerTest.js:176]: I am happy to see this test added in too.

**Review by kobryan0619 — APPROVED** [review]

**Pull request merged**